client/data/agency-dashboard/use-fetch-dashboard-sites.js

```diff
@@ -0,0 +1,35 @@
+import { useTranslate } from 'i18n-calypso';
+import { useQuery } from 'react-query';
+import { useDispatch } from 'react-redux';
+import { formatSites } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/utils';
+import { wpcomJetpackLicensing as wpcomJpl } from 'calypso/lib/wp';
+import { errorNotice } from 'calypso/state/notices/actions';
+
+/**
+ * Fetch all sites for the partner(agency)
+ *
+ * @returns {data, error, isLoading} Returns an object with the
+ * data associated to sites
+ */
+const useFetchDashboardSites = () => {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	return useQuery(
+		[ 'jetpack-cloud', 'agency-dashboard', 'sites' ],
+		() =>
+			wpcomJpl.req.get( {
+				path: '/jetpack-partner/dashboard/sites-mock',
+				apiNamespace: 'wpcom/v2',
+			} ),
+		{
+			refetchOnWindowFocus: false,
+			select: formatSites,
```

Comment:
I'm worried about `formatSites` doing more than it should and being useful only for the agency dashboard. I would expect a formatter to only format data A into shape B to facilitate its use, but this function is doing more. For instance, it adds functions to each site, and those functions return pieces of UI. As a result, `useFetchDashboardSites` will only work for this dashboard and not well for other features/components that don't require those pieces of UI.

Reply:
Totally makes sense. I have refactored this so that useFetchDashboardSites can be used at other places as well.